### PR TITLE
Add condition to checkDependencies when SourceRef is the same

### PR DESCRIPTION
**Pull request by HardBrainer**
As discussed in https://cloud-native.slack.com/archives/CLAJ40HV3/p1639665724208200 we identified a race condition in checkDependencies when using Receiver and both Kustomizations have the same SourceRef. 

If triggered by the Receiver the checkDependencies is executed before the dependent Kustomization applied the latest Revision. To prevent this we check if the LastAppliedRevision is already the new Revision.

In this Case deployment-demoapp has an dependsOn to init-demoapp. This time deployment-demoapp waits untils init-demoapp applied the latest Revision:

image

### CODE REVIEW

File: controllers/kustomization_controller.go [hunk 1]
```diff
@@ -490,6 +490,10 @@ func (r *KustomizationReconciler) checkDependencies(kustomization kustomizev1.Ku
 		if !apimeta.IsStatusConditionTrue(k.Status.Conditions, meta.ReadyCondition) {
 			return fmt.Errorf("dependency '%s' is not ready", dName)
 		}
+		
+		if len(k.Status.Conditions) == 0 || k.Spec.SourceRef.Name == kustomization.Spec.SourceRef.Name && k.Spec.SourceRef.Namespace == kustomization.Spec.SourceRef.Namespace && source.GetArtifact().Revision != k.Status.LastAppliedRevision {
```
Comment (stefanprodan) [inline controllers/kustomization_controller.go:494]: There is no need for `len(k.Status.Conditions) == 0` this is evaluated above. Another issue is that we only check the source name and namespace, but we really need to check the kind as well, as a GitRepository can have the same name as a Bucket.

Comment (HardBrainer) [reply controllers/kustomization_controller.go:494]: Fair enough i already thought about other specs to check but forgot the kind.

Comment (HardBrainer) [reply controllers/kustomization_controller.go]: I added the kind Check and remove the duplicated Conditions Check

**Review by stefanprodan — CHANGES_REQUESTED** [review]
Please run `make fmt` commit the changes then squash all commits into one. Thanks!

**Comment by HardBrainer** [comment]
Hey,

i performed a rebase on my branch and squashed all commits into one. `make fmt` did not create any new changes.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @HardBrainer 🏅

**Pull request merged**